app/javascript/components/Home.jsx

```diff
@@ -7,18 +7,39 @@ export default () => {
   // Table filters
   const [companyName, setCompanyName] = useState("");
   const [industry, setIndustry] = useState("");
-  const [minEmployee, setMinEmployee] = useState("");
-  const [minimumDealAmount, setMinimumDealAmount] = useState("");
+  const [minEmployee, setMinEmployee] = useState("0");
+  const [minimumDealAmount, setMinimumDealAmount] = useState("0");
+
+  const addQueryKeyToParams = (params) => {
+    return Object.fromEntries(
+      Object.entries(params).map(([key, value]) => {
+        return [`query[${key}]`, value];
+      })
+    );
+  }
+
+  const queryParams = () => {
+    return {
+      employee_count: minEmployee,
+      industry: industry,
+      name: companyName,
+      total_deal_amount: minimumDealAmount
+    }
+  }
 
   // Fetch companies from API
   useEffect(() => {
-    const url = "/api/v1/companies";
-    fetch(url)
-      .then((res) => {
-        return res.json();
-      })
-      .then((res) => setCompanies(res))
-  }, [])
+    const fetchCompanies = setTimeout(
+      async () => {
+        const url = `/api/v1/companies?${new URLSearchParams(addQueryKeyToParams(queryParams())).toString()}`;
+        const response = await fetch(url);
+        const json = await response.json();
+        setCompanies(json);
+      }, 500
+    )
+
+    return () => clearTimeout(fetchCompanies);
```

Comment:
The `setTimeout` and `clearTimeout` funcions are used to debounce the fetchRequest and avoid doing multiple fetch requests as the input changes. 

Lodash provides a debounce function as well that could be used